pytensor/sparse/basic.py

```diff
@@ -4260,3 +4256,82 @@ def grad(self, inputs, grads):
 
 
 construct_sparse_from_list = ConstructSparseFromList()
+
+
+class SparseBlockDiagonal(BaseBlockDiagonal):
+    __props__ = ("format",)
```

Comment:
Big gotcha you need to add the props of the base class as well

Reply:
I thought they get set union'd when you subclass?

Reply:
Maybe! Would be nice if we could do that but I don't know how.

We could also investigate using slots instead